testnet/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -0,0 +1,699 @@
+        let pkh = Hash160::from_data(&public_key.to_bytes()).to_bytes().to_vec();
+        let address = BitcoinAddress::from_bytes(
+            BitcoinNetworkType::Regtest,
+            BitcoinAddressType::PublicKeyHash,
+            &pkh)
+            .expect("Public key incorrect");        
+        let filter_addresses = vec![address.to_b58()];
+
+        let request_builder = self.get_rpc_request_builder();
+        let result = BitcoinRPCRequest::list_unspent(
+            request_builder,
+            filter_addresses.clone(), 
+            false, 
+            amount_required);
+
+        // Perform request
+        let mut utxos = match result {
+                Ok(utxos) => utxos,
+                Err(e) => {
+                    error!("Bitcoin RPC failure: error listing utxos {:?}", e);
+                    panic!();    
+                }
+        };
+
+        if utxos.len() == 0 {
+            let request_builder = self.get_rpc_request_builder();
+            let _result = BitcoinRPCRequest::import_public_key(
+                request_builder,
+                &public_key.to_hex());
+
+            // todo(ludo): rescan can take time. we should probably add a few retries, with exp backoff.
+            sleep_ms(1000);
+
+            let request_builder = self.get_rpc_request_builder();
+            let result = BitcoinRPCRequest::list_unspent(
+                request_builder,
+                filter_addresses, 
+                false, 
+                amount_required);
+
+            utxos = match result {
+                    Ok(utxos) => utxos,
+                    Err(e) => {
+                        error!("Bitcoin RPC failure: error listing utxos {:?}", e);
+                        panic!();    
+                    }
+            };
+
+            if utxos.len() == 0 {
+                return None
+            }
+        }
+
+        let total_unspent: u64 = utxos.iter().map(|o| o.get_sat_amount()).sum();
+        if total_unspent < amount_required {
+            return None
+        }
+
+        Some(utxos)
+    }
+
+    fn build_leader_key_register_tx(&mut self, payload: LeaderKeyRegisterPayload, signer: &mut BurnchainOpSigner) -> Option<Transaction> {
+        
+        let public_key = signer.get_public_key();
+
+        let (mut tx, utxos) = self.prepare_tx(&public_key, 0)?;
+
+        // Serialize the payload
+        let op_bytes = {
+            let mut buffer= vec![];
+            let mut magic_bytes = self.config.burnchain.magic_bytes.as_bytes().to_vec();
+            buffer.append(&mut magic_bytes);
+            payload.consensus_serialize(&mut buffer).expect("FATAL: invalid operation");
+            buffer
+        };
+
+        let consensus_output = TxOut {
+            value: 0,
+            script_pubkey: Builder::new()
+                .push_opcode(opcodes::All::OP_RETURN)
+                .push_slice(&op_bytes)
+                .into_script(),
+        };
+
+        tx.output = vec![consensus_output];
```

Comment:
I'm not seeing it here, but is there a change output added somewhere?  Also, is the transaction fee considered?